### PR TITLE
First cut of api doc

### PR DESCRIPTION
**Pull request by thelollies**
Very likely to need changes.

### CODE REVIEW

File: api.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+
+### <a name="team"></a>Team
+#### Representation
+```javascript
+{
+  kind:          "Team",
+  id:            {integer},  // Unique identifier
+  top_player:    {Player},   // Top player, see Table Layout
+  bottom_player: {Player}    // Bottom player, see Table Layout
```
Comment (aspett) [inline api.md]: Don't think it makes sense to have top_player and bottom_player. player1, player2 might be better as position isn't really important.

File: api.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+```
+#### Interface
+| Endpoint    | Verb     | Description |
+|:------------|----------|-------------|
+| `/teams/`   | `GET`    | List all teams
+| `/team/:id` | `GET`    | Show the specified team
+| `/teams/`   | `POST`   | Create a new team
+| `/team/:id` | `DELETE` | Delete the specified team
+
+Note Teams are immutable. If you wish to modify a team, delete and recreate it.
```
Comment (aspett) [inline api.md]: Whats the reasoning behind making a team immutable?

File: api.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+### <a name="game"></a>Game
+#### Representation
+```javascript
+{
+  kind:        "Game",
+  id:          {integer},     // Unique identifier
+  left_side:   {Team/Player}, // Team/Player on left, see Table Layout
+  right_side:  {Team/Player}, // Team/Player on right, see Table Layout
```
Comment (aspett) [inline api.md]: Do left and right side pertain to which side of the table the team _starts_ on?

**Comment by aspett** [comment]
Curious as to whether having left/right is important or not. Can't see any problems with just have team1, team2, player1, player2 etc?

File: api.md (hunk 1 above)
Comment (thelollies) [inline api.md]: I included it so you could show who is serving on the front end.

File: api.md (hunk 2 above)
Comment (thelollies) [inline api.md]: A Game holds two teams so if you change the players in a team you would change players' win/loss.

File: api.md (hunk 3 above)
Comment (thelollies) [inline api.md]: A team is always on the same side for the duration of a Game. They only swap between Games.

**Comment by thelollies** [comment]
Yeah the whole top/bottom thing purely exists so we can display player positions on the front end (you need to know who is on the left/right at a minimum so you know who's score should go up when the point won button is pressed).

File: api.md (hunk 2 above)
Comment (aspett) [inline api.md]: Hmm. Should team matches influence individual player statistics?

File: api.md (hunk 3 above)
Comment (aspett) [inline api.md]: Fair call

**Comment by aspett** [comment]
Fair call. :+1:

File: api.md (hunk 1 above)
Comment (aspett) [inline api.md]: :+1:

File: api.md (hunk 2 above)
Comment (thelollies) [inline api.md]: Unsure, it should be kept separate from singles stats for sure.

**Comment by aspett** [comment]
Must discuss match resource

File: api.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+}
+```
+#### Interface
+| Endpoint    | Verb     | Description |
+|:------------|----------|-------------|
+| `/games/`   | `GET`    | List all games
+| `/game/:id` | `GET`    | Show the specified game
+| `/games/`   | `POST`   | Create a new game
+| `/game/:id` | `DELETE` | Delete the specified game
+
```
Comment (aspett) [inline api.md]: Potentially want a `PUT /game/:id/point/{left|right}` to indicate a point added to the game, as we discussed.

**Pull request closed without merging**